### ISSUE TITLE
watchPath glob directory not working

### ISSUE THREAD

**Issue by nick297** [nextflow-io/nextflow#335]
I'm having issues with watchPath and glob when using . For example .watchPath( '/home/nick/files//.txt' ) gives the error: WARN: Cannot watch a not existing path: /home/nick/files/* -- Make sure that path exists and it is a directory

example code below
```
#!/usr/bin/env nextflow
txts=Channel
        .watchPath( 'files/**/*txt' )



process echoTest {

        input:
        val t from txts


        """
        echo ${t}
        """
}
```

**Comment by pditommaso** [comment]
I've pushed a patch for this problem and uploaded a new snapshot. You may want to give a try to it running: 

```
NXF_VER=0.24.4-SNAPSHOT nextflow run .. etc 
```

**Comment by nick297** [comment]
Thanks, I've given it a go with the above code, the error does not appear but the files are not picked up either

**Comment by pditommaso** [comment]
You are right, it's less trivial than I was thinking. The logic to recurse subdirectories need to be implemented. 
https://docs.oracle.com/javase/tutorial/essential/io/examples/WatchDir.java

**Comment by pditommaso** [comment]
I've uploaded a new snapshot that fixes the subdirectories problem. Run this command to update it: 

```
NXF_VER=0.24.4-SNAPSHOT CAPSULE_RESET=1 nextflow info
```

Then use NF with this command: 

```
NXF_VER=0.24.4-SNAPSHOT nextflow run .. etc 
```

**Comment by nick297** [comment]
This works great, thank you Paolo

**Comment by pditommaso** [comment]
Fixed with version `0.24.4` .
